player class added default movement

Roland-1124997 opened the pull request:
**Core Game Loop and Structure**

* Rewrote `main.py` to initialize the game and player, handle the main loop, process player actions, and update the display using the new `Game` and `Player` classes.
* Added `modules/Game.py` to encapsulate game state, screen management, grid calculations, event handling, and drawing the dungeon grid.

**Player**

* Added `modules/Player.py` to implement the `Player` class, including movement logic, input handling, sprite animation, and position management. The player uses a cooldown system and supports directional movement with animated frames.
* Introduced `modules/Player.json` to define animation frames for the player's idle and walking states in each direction, used by the `Player` class for sprite animation.

**Asset Loading**

* Added `utils/fileloader.py` with a `load_asset` function to extract frames from a sprite sheet,

shae-1046782 commented on main.py:
little nitpicky, but the formatting is off.

shae-1046782 commented on modules/Game.py:
would be cool to have some doctrings, comments, and type annotations

shae-1046782 commented on modules/Player.py:
again, for documentation purposes: docstrings, comments, and type annotations would be handy here

shae-1046782 reviewed (changes requested):
Code works, but it’s missing context. Please add comments, type annotations, and/or docstrings for readability.

Roland-1124997 commented:
# Summary
Comments have been resolved

## Added: 
- Type hints,
- Comments, 
- Docstrings  

### Refactor:
- Moved some logic into separate functions
- Marked some functions as private with _function_name
- Moved json structure into it`s own folder

shae-1046782 commented on modules/Player.py:
line 37 contains a `super().__init__()` function, but on line 19 `class Player:` doesn't seem to be inheriting from anything?

Roland-1124997 replied:
This is in preparation for calling the kill method for when the players hitpoints 
Reach zero

shae-1046782 commented on modules/Game.py:
suggestions (completely optional):
for lines 46, 51, 56, the `clock`, `screen`, and `deltatime`. would be best to use property decorators instead. ie:
```
@property
def screen(self) -> pygame.Surface:
    ...
```
